cmd/mt-index-cat/out/template_functions.go

```diff
@@ -0,0 +1,70 @@
+package out
+
+import (
+	"math/rand"
+	"strings"
+)
+
+// random choice between replacing a node with a wildcard, a char with a wildcard, and passthrough
+func pattern(in string) string {
+	mode := rand.Intn(3)
+	if mode == 0 {
+		// in this mode, replaces a node with a wildcard
+		parts := strings.Split(in, ".")
+		parts[rand.Intn(len(parts))] = "*"
+		return strings.Join(parts, ".")
+	} else if mode == 1 {
+		// randomly replace chars with a *
+		// note that in 1/5 cases, nothing happens
+		// and otherwise, sometimes valid patterns are produced,
+		// but it's also possible to produce patterns that won't match anything (if '.' was taken out)
+		chars := rand.Intn(5)
+		pos := rand.Intn(len(in) - chars)
```

Comment:
if `len(in)` is `<5` then it is possible that this will result be `rand.Intn(<=0)` (depending on `char`). That would panic according to: https://golang.org/pkg/math/rand/#Intn